hledger-lib/Hledger/Read/CsvReader.hs

```diff
@@ -724,14 +726,26 @@ transactionFromCsvRecord sourcepos rules record = t
                    _         -> "expenses:unknown"
     account1    = T.pack $ maybe "" render (mfieldtemplate "account1") `or` defaccount1
     account2    = T.pack $ maybe "" render (mfieldtemplate "account2") `or` defaccount2
-    balance     = maybe Nothing (parsebalance.render) $ mfieldtemplate "balance"
-    parsebalance str
+    balance'     = maybe Nothing (parsebalance "".render) $ mfieldtemplate "balance"
+    balance1'    = maybe Nothing (parsebalance "1".render) $ mfieldtemplate "balance1"
+    balance1     =
+      case (balance', balance1') of
+        (Nothing, Nothing)  -> Nothing
+        (balance, Nothing)  -> balance
+        (Nothing, balance1) -> balance1
+        (balance, balance1) ->
+          error' $    "both balance and balance1 have a value\n"
+                   ++ "    balance: "  ++ show balance ++ "\n"
+                   ++ "    balance1: " ++ show balance1 ++ "\n"
+                   ++ "    record: "     ++ showRecord record          
```

Comment:
I think this allows both balance and balance 1 to be used simultaneously as long as they don't both have a non-empty value in the same record. Ideally perhaps we could be more strict here, failing if both balance and balance1 are seen when parsing the rules.